pynxtools/dataconverter/readers/stm/stm_helper.py

```diff
@@ -0,0 +1,224 @@
+        new_temp_key = nexus_path.replace(field_to_replace, f"{field_to_replace}[{modified_field}]")
+        value = "value"
+        unit = "unit"
+        dict_orig_key_to_mod_key[nexus_path] = new_temp_key
+        if value in data_path:
+            path_to_data = data_path[value]
+            template[new_temp_key] = transform(data_dict[path_to_data]
+                                               if path_to_data in data_dict else None)
+        if unit in data_path:
+            path_to_data = data_path[unit]
+            template[new_temp_key + "/@units"] = transform(data_dict[path_to_data]
+                                                           if path_to_data in data_dict
+                                                           else None)
+
+
+def nested_path_to_slash_separated_path(nested_dict: dict,
+                                        flattened_dict: dict,
+                                        parent_path=''):
+    """Convert nested dict into slash separeted path upto certain level."""
+    start = '/'
+
+    for key, val in nested_dict.items():
+        path = parent_path + start + key
+        if isinstance(val, dict):
+            nested_path_to_slash_separated_path(val, flattened_dict, path)
+        else:
+            flattened_dict[path] = val
+
+
+def link_implementation(template, link_modified_dict):
+    """Rewrite the link compatible with hdf5 full path.
+    for e.g. convert /ENTRY[entry]/INSTRUMENT[instrument]/name to
+    /entry/instrument/name and rewrite in template.
+
+    Parameters
+    ----------
+    template : Template (dict)
+        To write out the hdf file
+    link_modified_dict : dict
+        The key corresponds to nxdl def path e.g. /ENTRY[entry]/INSTRUMENT[instrument]/NAME
+        and the value is the modified link path e.g.
+        /ENTRY[entry]/INSTRUMENT[special_instrument]/given_name where the
+        value is according to the implementaion of the NeXus def.
+    """
+    for _, val in template.items():
+        if isinstance(val, dict) and 'link' in val:
+            orig_link_path = val['link']
+            # Check whether any concept has been rewriten stored in key value
+            if orig_link_path in link_modified_dict:
+                # modified concepts come in a list together.
+                modif_link_hdf_path = convert_data_dict_path_to_hdf5_path(
+                    link_modified_dict[orig_link_path])
+                val['link'] = modif_link_hdf_path
+            else:
+                val['link'] = convert_data_dict_path_to_hdf5_path(orig_link_path)
+
+
+def cal_dx_by_dy(x_val: np.ndarray, y_val: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Calc conductance or gradiant dx/dy for x-variable and y-variable also return the result."""
+    dx_ = x_val[0::2] - x_val[1::2]
+    dy_ = y_val[0::2] - y_val[1::2]
+
+    dx_by_dy = dx_ / dy_
+
+    return dx_by_dy
+
+
+def cal_x_multi_x(x_val: np.ndarray, y_val: np.ndarray) -> np.ndarray:
```

Comment:
Maybe here an in the next function a function call is overkill or are you expecting that in the future the code in each function will get longer/more?